### PR TITLE
Log form values

### PR DESCRIPTION
**Pull request by mgwalker**
Per @nkkl's request, record the form values as they're being entered by the user.  Some notes:

* the API stores the form values in memory, so if the API restarts for some reason, the collected values will be reset
* the form values are stored in a JSON object, keyed on the timestamp of when the user first began interacting with the form
* if the user reloads their browser session, they'll get a new timestamp and they'll be represented twice in the logged results (I tried to get around this by stashing a cookie but it refused to work and I didn't want to spend any more time on it)
* logging is enabled or disabled at ***build*** time, using the `LOG_FORM_INTERACTIONS` environment variable - by default this is `false`, but if it is set to `true`, form interactions will be logged
* the API URL is inserted into the code at build time; the build process will replace occurrences of `process.env.API_URL` with the actual constant value stored in that environment variable
* I updated the deployment scripts to set the environment variables as appropriate (enabled for UX testing, disabled for staging)

Here's a sample of what the JSON looks like (please forgive the not-great form element names, they're interim!):

```json
{
  "Mon Feb 05 2018 11:33:04 GMT-0600 (CST)": {
    "stateStart": {
      "name": "Denise Nagelschmidt",
      "position": "Health Reform Portfolio Director",
      "email": "denise.nagelschmidt@vermont.gov",
      "phone": "802-879-5900",
      "stateName": "vt"
    },
    "stateContacts": {
      "medicaidOffice": {
        "address1": "Department of Vermont Health Access",
        "address2": "280 State Drive",
        "city": "Waterbury",
        "state": "Vermont",
        "zip": "05671-1010"
      },
      "medicaidDirector": {
        "name": "Cory Gustafson",
        "email": "cory.gustafson@vermont.gov",
        "phone": "802-879‐5901"
      }
    },
    "apdOverview": {
      "vision": "a vision statement here",
      "benefits": "and some benefits, too"
    },
    "form3": {
      "activities": [
        "administration",
        "strategic-planning",
        "outreach",
        "design-and-development"
      ]
    },
    "personnel": {
      "state": [
        {
          "name": "Alicia Axelrod",
          "jobTitle": "big cheese",
          "nextCompensation": 78400,
          "nextTime": 50,
          "nextNextTime": 30
        }
      ],
      "contracting": [
        {
          "name": "Barbara Bigelow",
          "jobTitle": "software engineer",
          "nextCompensation": 89324,
          "nextTime": 80,
          "nextNextTime": 80
        },
        {
          "name": "Chandra Cogsworth",
          "jobTitle": "content designer",
          "nextCompensation": 91535,
          "nextTime": 70,
          "nextNextTime": 80
        }
      ]
    }
  }
}
```

For now, logging is disabled by setting a constant at build time and then evaluating it at runtime.  Longer term, it would probably be nice to [use `IgnorePlugin` and `UglifyJSPlugin` to completely remove the logging module](https://remarkablemark.org/blog/2017/02/25/webpack-ignore-module/) at build time, so there aren't any runtime checks at all.

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- ~~The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented~~ N/A
- [x] The change has been documented

### This feature is done when...
- ~~Design has approved the experience~~ N/A
- [x] Product has approved the experience (@nkkl)

### CODE REVIEW

**Comment by nkkl** [comment]
Looks good to me!

File: web/src/util/formLogger.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      axios
+        .post(`${apiURL}/log-form`, { user, form: formValues })
+        .then(() => { timeout = null; })
+        .catch(() => { timeout = null; });
+    }, 2000);
+  }
+  return null;
+};
+
+const mapStateToProps = state => ({
```
Comment (brendansudol) [inline web/src/util/formLogger.js]: one-line equivalent utilizing destructuring and object literal property value shorthand:

const mapStateToProps = ({ form }) => ({ form })

**Comment by brendansudol** [comment]
love the documentation additions, btw!

File: web/src/util/formLogger.js (hunk 1 above)
Comment (mgwalker) [reply web/src/util/formLogger.js]: Love it.  Going to make the change shortly.

**Pull request merged**